roxie/ccd/ccdserver.cpp

```diff
@@ -510,23 +506,22 @@ class CRoxieServerActivityFactoryBase : public CActivityFactory, implements IRox
     virtual void getNodeProgressInfo(IPropertyTree &node) const
     {
         CActivityFactory::getNodeProgressInfo(node);
-        CriticalBlock b(statsCrit);
-        if (started)
-            putStatsValue(&node, "_roxieStarted", "sum", started);
+        auto _started = started.load(std::memory_order_relaxed);
+        if (_started)
+            putStatsValue(&node, "_roxieStarted", "sum", _started);
     }
 
     virtual void resetNodeProgressInfo()
     {
         CActivityFactory::resetNodeProgressInfo();
-        CriticalBlock b(statsCrit);
-        processed = 0;
-        started = 0;
+        processed.store(0, std::memory_order_relaxed);
+        started.store(0, std::memory_order_relaxed);
     }
     virtual void getActivityMetrics(StringBuffer &reply) const
     {
         CActivityFactory::getActivityMetrics(reply);
+        putStatsValue(reply, "_roxieStarted", "sum", started.load(std::memory_order_relaxed));
         CriticalBlock b(statsCrit);
```

Comment:
It might be good to eliminate this critsec altogether - I suspect the RuntimeStatisticGatherer could be made sufficiently threadsafe using atomics. The critsec is now only used here and in mergeStats.